### PR TITLE
refactor: introduce cards view model (which contains results by status in a property)

### PR DESCRIPTION
**Pull request by waabid**
#### Description of changes

Previously we had only card related data (in properties like: ruleResultsByStatus). We will now be adding additional data, such as whether the visual helper toggle is enabled or disabled and whether all rule groups are expanded; the former is the primary motivator to make this change.

Most of these changes are structuring the data to this new type.

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [n/a] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). Check workflow guide at: `<rootDir>/docs/workflow.md`
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/common/rule-based-view-model-provider.ts [hunk 1]
```diff
@@ -9,20 +9,21 @@ import {
...
-export type GetUnifiedRuleResultsDelegate = (
+export type GetCardViewData = (
     rules: UnifiedRule[],
     results: UnifiedResult[],
     cardSelectionViewData: CardSelectionViewData,
-) => CardRuleResultsByStatus;
+) => CardsViewModel;
 
-export const getUnifiedRuleResults: GetUnifiedRuleResultsDelegate = (
+export const getUnifiedRuleResults: GetCardViewData = (
```
Comment (DaveTryon) [inline src/common/rule-based-view-model-provider.ts]: Should this method be renamed to getCardViewData or something similar?

Comment (waabid) [reply src/common/rule-based-view-model-provider.ts]: Updated

File: src/common/components/cards/result-section.tsx [hunk 2]
```diff
@@ -4,8 +4,6 @@ import { css } from '@uifabric/utilities';
 import { NamedFC } from 'common/react/named-fc';
 import * as React from 'react';
 
-import { TargetAppData } from '../../../common/types/store-data/unified-data-interface';
```
Comment (DaveTryon) [inline src/common/components/cards/result-section.tsx]: Edits in this file seems unrelated to the rest of the change. Opportunistic cleanup or is it actually needed?

Comment (waabid) [reply src/common/components/cards/result-section.tsx]: The former; it was a bit confusing why these properties were declared twice, effectively speaking.
